### DISCUSSION TITLE
Consider implementing PSR-7 & PSR-15

### DISCUSSION THREAD

**Discussion by rahmanramsi** [pkp/pkp-lib#9962] [Proposals]
With implementing PSR-7 & PSR-15, this will allow application to be running on new runtime (Swoole, RoadRunner, FrankenPHP, etc). 
The application will gain a huge performance boost with this implementation.

**Comment by asmecher** [comment]
@rahmanramsi, I support this, but the path to reaching PSR-7 and PSR-15 compliance will definitely be to continue moving OJS into a Laravel platform. We have recently moved API request handling over to Laravel, for example, and starting with 3.5 [session management](https://github.com/pkp/pkp-lib/issues/9566) is migrated too. We're gradually replacing the old homebrew app stack with Laravel, and that'll give us PSR-7 and PSR-15 in the process.
